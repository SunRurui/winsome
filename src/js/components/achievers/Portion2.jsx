import React from 'react';
<<<<<<< HEAD
import { shape, arrayOf, string } from 'prop-types';
=======
import { shape, arrayOf, string, number } from 'prop-types';
>>>>>>> 0e2bb0bf8a7c4da622c6eee85d4f822541b0cb07
import Part from './Part';

class Portion2 extends React.Component {
	renderArrayHandler() {
		const { achieversData } = this.props;
		return (
			<div>
				<h2>{achieversData.class16Data.textYear}</h2>
				<div className="box" />
				{achieversData.class16Data.data.map(item => (
					<Part
<<<<<<< HEAD
						key={Math.random()}
=======
						key={item.id}
>>>>>>> 0e2bb0bf8a7c4da622c6eee85d4f822541b0cb07
						text={item.text}
						textHeading={item.textHeading}
						image={item.image}
					/>
				))}
			</div>
		);
	}
	render() {
		return (
			<div className="portion2">
				{this.renderArrayHandler()}
				<div className="clear" />
			</div>
		);
	}
}

export default Portion2;

Portion2.propTypes = {
	achieversData: shape({
		firstPart: shape({
			heading: string.isRequired,
			text: string.isRequired
		}).isRequired,
		class16Data: shape({
			textYear: string.isRequired,
			data: arrayOf(
				shape({
<<<<<<< HEAD
=======
					id: number,
>>>>>>> 0e2bb0bf8a7c4da622c6eee85d4f822541b0cb07
					textHeading: string.isRequired,
					text: string.isRequired,
					image: string.isRequired
				})
			).isRequired
		}).isRequired
	}).isRequired
};
