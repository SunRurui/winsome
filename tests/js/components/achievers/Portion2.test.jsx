import React from 'react';
import Enzyme,{shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Portion2 from '../../../../src/js/components/achievers/Portion2'

Enzyme.configure({adapter:new Adapter()})

describe('Test the Portion2 Component',()=>{

	const achieversData={
	firstPart:{
		heading:'',
		text:''
		
	},
	class17Data:{
		textYear:'',
		data:[
			{
				textHeading:'',
				text:'',
				image:''
			},
			{
				textHeading:'Ahmed',
				text:'',
				image:''
			},
			{
				textHeading:'Waqar',
				text:'',
				image:''
			},
			{
				textHeading:'Ahsan',
				text:'',
				image:''
			},
			{
				textHeading:'Hassan',
				text:'',
				image:''
			}
		]
	},
	class16Data:{
		textYear:'',
		data:[
			{
				textHeading:'Rehan',
				text:'',
				image:''
			},
			{
				textHeading:'Taimoor',
				text:'',
				image:''
			},
			{
				textHeading:'Mushraf',
				text:'',
				image:''
			}
		]
	}
}

	const wrapper=shallow(<Portion2 achieversData={achieversData} />)
	it('test the rendring',()=>{
		expect(wrapper).toMatchSnapshot();
	})
	it('test that the data is an object',()=>{
		expect('object').toBe(typeof(achieversData))
	})
})