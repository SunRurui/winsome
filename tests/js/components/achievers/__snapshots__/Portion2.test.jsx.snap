// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Portion2 Component test the rendring 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Portion2
    achieversData={
      Object {
        "class16Data": Object {
          "data": Array [
            Object {
              "image": "",
              "text": "",
              "textHeading": "Rehan",
            },
            Object {
              "image": "",
              "text": "",
              "textHeading": "Taimoor",
            },
            Object {
              "image": "",
              "text": "",
              "textHeading": "Mushraf",
            },
          ],
          "textYear": "",
        },
        "class17Data": Object {
          "data": Array [
            Object {
              "image": "",
              "text": "",
              "textHeading": "",
            },
            Object {
              "image": "",
              "text": "",
              "textHeading": "Ahmed",
            },
            Object {
              "image": "",
              "text": "",
              "textHeading": "Waqar",
            },
            Object {
              "image": "",
              "text": "",
              "textHeading": "Ahsan",
            },
            Object {
              "image": "",
              "text": "",
              "textHeading": "Hassan",
            },
          ],
          "textYear": "",
        },
        "firstPart": Object {
          "heading": "",
          "text": "",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <h2>
            
          </h2>
          <div
            className="box"
          />
          <Part
            image=""
            text=""
            textHeading="Rehan"
          />
          <Part
            image=""
            text=""
            textHeading="Taimoor"
          />
          <Part
            image=""
            text=""
            textHeading="Mushraf"
          />
        </div>,
        <div
          className="clear"
        />,
      ],
      "className": "portion2",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              
            </h2>,
            <div
              className="box"
            />,
            Array [
              <Part
                image=""
                text=""
                textHeading="Rehan"
              />,
              <Part
                image=""
                text=""
                textHeading="Taimoor"
              />,
              <Part
                image=""
                text=""
                textHeading="Mushraf"
              />,
            ],
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "box",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "image": "",
              "text": "",
              "textHeading": "Rehan",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "image": "",
              "text": "",
              "textHeading": "Taimoor",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "class",
            "props": Object {
              "image": "",
              "text": "",
              "textHeading": "Mushraf",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "clear",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <h2>
              
            </h2>
            <div
              className="box"
            />
            <Part
              image=""
              text=""
              textHeading="Rehan"
            />
            <Part
              image=""
              text=""
              textHeading="Taimoor"
            />
            <Part
              image=""
              text=""
              textHeading="Mushraf"
            />
          </div>,
          <div
            className="clear"
          />,
        ],
        "className": "portion2",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                
              </h2>,
              <div
                className="box"
              />,
              Array [
                <Part
                  image=""
                  text=""
                  textHeading="Rehan"
                />,
                <Part
                  image=""
                  text=""
                  textHeading="Taimoor"
                />,
                <Part
                  image=""
                  text=""
                  textHeading="Mushraf"
                />,
              ],
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "box",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "image": "",
                "text": "",
                "textHeading": "Rehan",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "image": "",
                "text": "",
                "textHeading": "Taimoor",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "class",
              "props": Object {
                "image": "",
                "text": "",
                "textHeading": "Mushraf",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "clear",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
